web/ui/src/Services/ServiceDetailsController.js

```diff
@@ -499,7 +499,8 @@
 
             $modalService.create({
                 template: $translate.instant("remove_public_endpoint") + ": <strong>"+
-                          (publicEndpoint.Name ? publicEndpoint.Name : "port " + publicEndpoint.PortAddr) + "</strong>",
+                          (publicEndpoint.Name ? publicEndpoint.Name : "port " + publicEndpoint.PortAddr) + "</strong><br><br>"+
+                          "The <strong>" + publicEndpoint.Application + "</strong> service will be restarted.",
```

Comment:
The action that is going to occur should be very explicitly spelled out here. Something like "After the public endpoint is removed, the HMaster service will automatically be restarted."